libraries/azure_policy_insights_query_results.rb

```diff
@@ -0,0 +1,77 @@
+require 'azure_generic_resources'
+
+class AzurePolicyInsightsQueryResults < AzureGenericResources
+  name 'azure_policy_insights_query_results'
+  desc 'Lists a collection of Azure Policy Insights Query Results'
+  example <<-EXAMPLE
+    describe azure_policy_insights_query_results(policy_definition: 'de875639-505c-4c00-b2ab-bb290dab9a54') do
+      it { should exist }
+    end
+  EXAMPLE
+  attr_reader :table
+
+  def initialize(opts = {})
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+
+    # this is set so that we can run validations
+    # new @opts will be set inside parent classes
+    @opts = opts
+    validate_parameters(required: %i(policy_definition), allow: %i(filter_free_text))
+    resource_provider = specific_resource_constraint('Microsoft.Authorization/policyDefinitions', opts)
+    opts[:resource_uri] = ['providers',  resource_provider, opts.delete(:policy_definition),
+                           'providers/Microsoft.PolicyInsights/policyStates/latest/queryResults'].join('/')
+    opts[:add_subscription_id] = true
+    opts[:method] = 'post'
+
+    super
+
+    return if failed_resource?
+
+    table_schema = [
```

Comment:
Hi @sathish-progress 
It might be worth exposing all the fields we get from Azure. An example of missing fields that are actually quite useful:-

- complianceReasonCode (sometimes populated with a failure type)
- policyAssignmentParameters (these are sometimes overridden from default in sub scopes)
- policyDefinitionAction (which tells us things like whether this is a DINE or AINE policy)
- timestamp (which in my MVP resource I convert to a Ruby Time class for more useful filtering)

Reply:
yeah make sense, I will add those